presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/IndexJoinOptimizer.java

```diff
@@ -330,7 +330,7 @@ public PlanNode visitWindow(WindowNode node, RewriteContext<Context> context)
             // Only RANGE frame type currently supported for aggregation functions because it guarantees the
             // same value for each peer group.
             // ROWS frame type requires the ordering to be fully deterministic (e.g. deterministically sorted on all columns)
-            if (node.getFrame().getType() != WindowFrame.Type.RANGE) {
+            if (node.getFrames().values().stream().anyMatch(frame -> frame.getType() != WindowFrame.Type.RANGE)) { // TODO: extract frames of type RANGE and allow optimization on them
```

Comment:
As you wish, here's fixup: da5a6e1411fdb25b8c35124b33101804d190404c.
Again, will rebase to apply properly between Eric's and my changes since his commits interfere with that change slightly.